importers/ableroad.com/easy_importer.py

```diff
@@ -0,0 +1,226 @@
+
+    yelprating = dombus.xpath('.//img[@class="yelprating"]/@alt')
+    yelprating = getyelp(yelprating)
+    row.append(yelprating)
+
+    address = dombus.xpath('.//address/text()')
+
+    street = getstreet(address)
+    row.append(street)
+
+    city = getcity(address)
+    row.append(city)
+
+    state = getstate(address)
+    row.append(state)
+
+    postcode = getpostcode(address)
+    row.append(postcode)
+
+    phone = getphone(address)
+    row.append(phone)
+
+    latlng = dombus.getprevious().text #lat lng is hidden in javascript
+    latlng = getlatlng(latlng)
+    row = row + latlng
+    return row
+
+def write_csv(rows):
+    csv_filename = 'export.csv'
+    if os.path.exists(csv_filename):
+        csv_file = open(csv_filename,'a')
+        writer = csv.writer(csv_file, delimiter=',',quotechar='"', quoting=csv.QUOTE_ALL)
+    else:
+        csv_file = open(csv_filename,'w')
+        writer = csv.writer(csv_file, delimiter=',',quotechar='"', quoting=csv.QUOTE_ALL)
+        title = ['name','category','distance','neighborhood','ablerating','yelprating','street','city',
+                'state','postcode','phone','lat','lng']
+        writer.writerow(title)
+    for row in rows:
+        writer.writerow(row)
+    csv_file.close()
+    logging.info('csv file has been written!')
+
+def download_if_not(url, islocalfile):
+    pagecontent = ''
+
+    if islocalfile and os.path.exists('tmp.html'):
+        f = open('tmp.html','r')
+        pagecontent = f.read()
+    else:
+        page = requests.get(url)
+        f = open('tmp.html','w')
+        f.write(page.content)
+        pagecontent = page.content
+
+    f.close()
+    return pagecontent
+
+def gen_url():
+    locations = ['windsor, ontario']
+    for location in locations:
+            for category_id in range(2, 22):
+                for page in range(0,10):
+                    offset = page * 20
+                    url = 'http://ableroad.com/search.php?s=&s1=' + urllib2.quote(location) + \
+                            '&cat=' + str(category_id) + '&offset=' + str(offset) + '&action=search'
+                    logging.info(url)
+                    mainapp(url)
+                    time.sleep(5)
+
+def mainapp(url, islocalfile=False):
+    dom = html.fromstring(download_if_not(url, islocalfile))
+    businesses = dom.xpath('//div[@class="bigresultframe"]')
+    rows = []
+    for bus in businesses:
+        row = extract_info(bus)
+        rows.append(row)
+    write_csv(rows)
+
+def setuplogging():
+    #logging setting
+    logging.basicConfig(filename='importer.log',level=logging.DEBUG,\
+            format='%(asctime)s - %(name)s - %(levelname)s - %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p')
+
+def test_download():
+    url = 'http://ableroad.com/search.php?s=reds-kitchen-tavern&offset=0&s1=peabody+ma&searchBut=Search&cat=1&action=search'
+    mainapp(url, True)
+
+def main():
+    setuplogging()
+    gen_url()
+    #test_download()
+
+main()
```

Comment:
Put an if-statement around this main() call.
```
if __name__ == '__main__':
   main()
```
Unlike a c or c++ program, Python code will run immediately and executing code unconditionally like this restricts developers from importing your module and using it as a helper to some other application they may be writing.  The suggested if-statement will checks "if not running/importing as helper module for another app, run main()."